mlflow/tensorflow/_autolog.py

```diff
@@ -52,11 +59,162 @@ def on_epoch_end(self, epoch, logs=None):
+    else:
+        warnings.warn(
+            f"Tensorflow estimator doesn't support '{type(input_features)}' type for features"
+        )
+    return input_feature_slice
+
+
+def _extract_sample_numpy_dict(
+    input_numpy_features_dict: Dict[str, np.ndarray]
+) -> Union[Dict[str, np.ndarray], np.ndarray]:
+    """
+    Extracts `INPUT_EXAMPLE_SAMPLE_ROWS` sample from next_input
+    as numpy array of dict(str -> ndarray) type.
+
+    :param input_numpy_features_dict: A tensor or numpy array
+    :return:a slice (limit `INPUT_EXAMPLE_SAMPLE_ROWS`)  of the input of same type as next_input
+
+    Examples
+    --------
+    when next_input is dict:
+    >>> input_data = {"a": np.array([1, 2, 3, 4, 5, 6, 7, 8])}
+    >>> _extract_sample_numpy_dict(input_data)
+    {'a': array([1, 2, 3, 4, 5])}
+
+    """
+    sliced_data_as_numpy = None
+    if isinstance(input_numpy_features_dict, dict):
+        sliced_data_as_numpy = {
+            k: _extract_input_example_from_tensor_or_ndarray(v)
+            for k, v in input_numpy_features_dict.items()
+        }
+    return sliced_data_as_numpy
+
+
+def _extract_input_example_from_batched_tf_dataset(
+    dataset: tensorflow.data.Dataset,
+) -> Union[np.ndarray, Dict[str, np.ndarray]]:
+    """
+    Extracts sample feature tensors from the input dataset as numpy array.
+    Input Dataset's tensors must contain tuple of (features, labels) that are
+    used for tensorflow/keras train or fit methods
+
+
+    :param dataset: a tensorflow batched/unbatched dataset representing tuple of (features, labels)
+    :return: a numpy array of length `INPUT_EXAMPLE_SAMPLE_ROWS`
+
+    Examples
+    --------
+    >>> input_dataset = tensorflow.data.Dataset.from_tensor_slices(
+    ...                 ({'SepalLength': np.array(list(range(0, 20))),
+    ...                 'SepalWidth': np.array(list(range(0, 20))),
+    ...                 'PetalLength': np.array(list(range(0, 20))),
+    ...                 'PetalWidth': np.array(list(range(0, 20)))},
+    ...                 np.array(list(range(0, 20))))).batch(10)
+    >>> _extract_input_example_from_batched_tf_dataset(input_dataset)
+    {'SepalLength': array([0, 1, 2, 3, 4]),
+    'SepalWidth': array([0, 1, 2, 3, 4]),
+    'PetalLength': array([0, 1, 2, 3, 4]),
+    'PetalWidth': array([0, 1, 2, 3, 4])}
+
+    """
+    limited_df_iter = list(dataset.take(INPUT_EXAMPLE_SAMPLE_ROWS))
+    first_batch = limited_df_iter[0]
+    if isinstance(first_batch, tuple):
+        features = first_batch[0]
+        if isinstance(features, dict):
+            return _extract_sample_numpy_dict(features)
+        elif isinstance(features, (np.ndarray, tensorflow.Tensor)):
+            return _extract_input_example_from_tensor_or_ndarray(features)
+        raise TypeError(f"Unsupported type for features of type: {type(features)}")
+    raise TypeError(f"Unsupported type for dataset batch slice of type: {type(first_batch)}")
```

Comment:
@bali0019 We shouldn't throw here; this will cause autologging to short-circuit.